vital/tests/test_image.cxx

```diff
@@ -36,6 +37,33 @@ value_at( unsigned int i, unsigned int j, unsigned int k )
   return static_cast<byte>( ( ( ks * k ) + ( js * j ) + ( is * i ) ) % 255 );
 }
 
+// ----------------------------------------------------------------------------
+void check_equal( image_memory mem_1, image_memory mem_2 )
+{
+  char* mem_1_data = (char*) mem_1.data();
+  char* mem_2_data = (char*) mem_2.data();
+
+  EXPECT_EQ( mem_1.size(), mem_2.size() );
+  for(unsigned int i=0; mem_1_data[i]!='\0'; i++)
```

Comment:
These are not actually null terminated strings, so you don't want to loop until you hit a null character you want to iterate through `mem_1.size()` bytes.